partner_event/models/res_partner.py

```diff
@@ -12,20 +14,43 @@ class ResPartner(models.Model):
     registrations = fields.One2many(
         string="Event registrations",
         comodel_name='event.registration', inverse_name="partner_id")
+    event_count = fields.Integer(
+        string='Events',
+        compute='_compute_event_count',
+        help="Count of events with confirmed registrations.",
+    )
     registration_count = fields.Integer(
-        string='Event registrations number', compute='_count_registration',
+        string='Event registrations number',
+        compute='_compute_registration_count',
         store=True)
     attended_registration_count = fields.Integer(
         string='Event attended registrations number',
-        compute='_count_attended_registration', store=True)
+        compute='_compute_attended_registration_count', store=True)
 
-    @api.one
+    @api.multi
     @api.depends('registrations')
-    def _count_registration(self):
-        self.registration_count = len(self.registrations)
+    def _compute_registration_count(self):
+        for partner in self:
+            partner.registration_count = len(partner.registrations)
 
-    @api.one
+    @api.multi
+    def _compute_event_count(self):
+        for partner in self:
+            partner.event_count = len(
+                self.env["event.registration"].search([
```

Comment:
The problem here is that a partner can have multiple registrations for the same event. Thus the `mapped("event_id")` is used on the obtained records to obtain the unique count.